tds_go113_test.go

```diff
@@ -0,0 +1,93 @@
+//go:build go1.13
+// +build go1.13
+
+package mssql
+
+import (
+	"database/sql"
+	"errors"
+	"net"
+	"testing"
+
+	"github.com/microsoft/go-mssqldb/msdsn"
+)
+
+// TestConnectError tests wrapped errors from connection establishing. It uses
+// error handling introduced in Go 1.13, that's the reason for conditional test.
+func TestConnectError(t *testing.T) {
+	loadConnParams := func(t *testing.T) msdsn.Config {
+		params := testConnParams(t)
+		if params.Encryption == msdsn.EncryptionRequired {
+			t.Skip("Unable to test connection to IP for servers that expect encryption")
+		}
+
+		if params.Host == "." {
+			params.Host = "127.0.0.1"
+		} else {
+			ips, err := net.LookupIP(params.Host)
+			if err != nil {
+				t.Fatal("Unable to lookup IP", err)
+			}
+			params.Host = ips[0].String()
+		}
+		return params
+	}
+	connAndPing := func(t *testing.T, params msdsn.Config) error {
+		connStr := params.URL().String()
+		conn, err := sql.Open("mssql", connStr)
+		if err != nil {
+			t.Fatal("Open connection failed:", err.Error())
+			return nil
+		}
+		pingErr := conn.Ping()
+		if pingErr == nil {
+			t.Fatal("Error required")
+			return nil
+		}
+		return pingErr
+	}
+	t.Run("bad port - refused connection", func(t *testing.T) {
+		params := loadConnParams(t)
+		// port where nothing listens on. Port 666 is reserved for Doom multiplayer
+		// server, hopefully no-one runs one in CI or in development environment.
+		params.Port = 666
+
```

Comment:
you need to clear Instance too

Reply:
AFAIK, the port shouldn't be resolved if specified, no? I remember this, because we used to struggle with the bug.

https://github.com/denisenkom/go-mssqldb/issues/714
https://github.com/microsoft/go-mssqldb/pull/6

Reply:
the failure is line 83 below
If you don't care about what's in the test environment variables or config file, don't use loadConnParams, just hard code something that will fail.

Reply:
I may need to fix `CallBrowser` in protocol.go to ignore `Instance` when `Port != 0`

Reply:
@shueybubbles 
Actually we use both, in case we use SSH tunneling/port forwarding. The port is the local forwarded port, but the instance name is required sometimes too. I'm not that familiar with SQL server, and also it's quite off topic for this PR :P

Reply:
Please open an issue if I've broken a connectivity scenario related to ports and instances. AFAIK the instance name for SQL Server is only used to look up a port or named pipe from the SQL Browser service. Most modern deployments disable the browser service and require the client to specify the port when not using the default.

Reply:
> Please open an issue if I've broken a connectivity scenario related to ports and instances. AFAIK the instance name for SQL Server is only used to look up a port or named pipe from the SQL Browser service. Most modern deployments disable the browser service and require the client to specify the port when not using the default.

It's French server :grimacing: will let you know if it breaks